reframe/core/modules.py

```diff
@@ -105,21 +106,19 @@ def resolve_module(self, name):
         :raises: :class:`reframe.core.exceptions.ConfigError` if the mapping
             contains a cycle.
         """
-        ret = []
+        ret = OrderedSet()
         visited = set()
         unvisited = [(name, None)]
         path = []
-        loop_nodes = set()
         while unvisited:
             node, parent = unvisited.pop()
             # Adjust the path
             while path and path[-1] != parent:
                 path.pop()
 
             # Handle modules mappings with self loops
-            if node == parent and node not in loop_nodes:
-                loop_nodes.add(node)
-                ret.append(node)
+            if node == parent and node not in ret:
```

Comment:
You don't need the `not in ret` part. It's handled the `OrderedSet`.

Reply:
A ok, I thought that it would update the order if it was already in.